server/routes/category.js

```diff
@@ -32,6 +34,21 @@ router.get('/getAll', async (req, res) => {
         res.status(500).json({message: error.message})
     }
 })
+
+// Get by Name Method
+router.get('/getOne/:name', async (req, res) => {
+    try{
+        const data = await categoryServices.getCategoryByName(req.params.name);
+        data["visualizations"] = await Promise.all(data["visualizations"].map(
+            visualizationServices.getVisualizationByTitle
+        ));
+        res.json(data)
+    }
+    catch(error){
+        res.status(500).json({message: error.message})
+    }
+})
+
```

Comment:
Delete by ID isn't part of this PR, but we can remove the delete in the route name since these routes are already prefixed.